backend/cn/check.ml

```diff
@@ -531,6 +530,21 @@ let rec check_pexpr (pe : BT.t mu_pexpr) (k : IT.t -> unit m) : unit m =
        | ty -> fail (fun _ -> {loc; msg = Mismatch {has = BT.pp ty; expect = !^"comparable type"}})
      in
      begin match op with
+     | OpDiv ->
+        let@ () = WellTyped.ensure_base_type loc ~expect (bt_of_pexpr pe1) in
+        let@ () = WellTyped.ensure_bits_type loc expect in
+        let@ () = WellTyped.ensure_bits_type loc (bt_of_pexpr pe2) in
```

Comment:
This check should ensure that the rhs has the same type as the lhs. Right now the code accepts the case where they are both bits but of different size.